version: "3"
services:
  cache_server1:
    build: .
    ports:
      - 9527:80
    deploy:
      restart_policy:
        condition: on-failure
    networks:
      server:
          ipv4_address: 192.24.0.2
    command: ["python3", "cache_server.py", "192.24.0.2", "192.24.0.3", "192.24.0.4"]  # pyhton3 cache_server.py {my_ip} {nearby_ip} {nearby_ip} {nearby_ip}

  cache_server2:
    build: .
    ports:
      - 9528:80
    deploy:
      restart_policy:
        condition: on-failure
    networks:
      server:
          ipv4_address: 192.24.0.3
    command: ["python3", "cache_server.py", "192.24.0.3", "192.24.0.2", "192.24.0.4"]
 
  cache_server3:
    build: .
    ports:
      - 9529:80
    networks:
      server:
          ipv4_address: 192.24.0.4
    deploy:
      restart_policy:
        condition: on-failure
    command: ["python3", "cache_server.py", "192.24.0.4", "192.24.0.2", "192.24.0.3"]

networks:
  server:
    driver:
      bridge
    ipam:
      config:
        - subnet: 192.24.0.0/24
          gateway: 192.24.0.1